api/src/Page/DC.php

```diff
@@ -177,7 +177,12 @@ function _data_collections($single = null)
         $info = array();
         # Visits
         if ($this->has_arg('visit')) {
-            $info = $this->db->pq("SELECT TO_CHAR(s.startdate, 'HH24') as sh, TO_CHAR(s.startdate, 'DDMMYYYY') as dmy, s.sessionid, s.beamlinename as bl FROM blsession s INNER JOIN proposal p ON (p.proposalid = s.proposalid) WHERE CONCAT(p.proposalcode, p.proposalnumber, '-', s.visit_number) LIKE :1", array($this->arg('visit')));
+            $pattern = '/([A-z]+)(\d+)-(\d+)/';
+            preg_match($pattern, $this->arg('visit'), $matches);
+            if (!sizeof($matches))
```

Comment:
This is caught and displayed as "no such proposal", but shouldn't this check that the number of matches is equal to three, rather than greater than 0? 

Right now you could type in "mx12345" and it would be a success

Reply:
No, I think the regex only matches if all the regex matches, and then it is portioned into groups afterwards. So the size is always 0 or 4 ($matches includes the whole match as well as the groups)
```
[ndg63276@i03-ws001 ~]$ php -a
Interactive shell
php > function get_matches($visit) {
php {     $pattern = '/([A-z]+)(\d+)-(\d+)/';
php {     preg_match($pattern, $visit, $matches);
php {     echo(sizeof($matches));
php { }
php > 
php > get_matches('mx12345-1');
4
php > get_matches('mx12345');
0

```

Reply:
Oh yeah, I noticed that what I said made no sense and resolved the conversation because of it, but now I wonder if this is a redundant check, since `$arg_list` checks for it anyway... It isn't hurting anyone, so I guess it doesn't matter